qml/pages/CameraUI.qml

```diff
@@ -22,79 +23,48 @@ Page {
+            if (reading.orientation >= OrientationReading.TopUp && reading.orientation <= OrientationReading.RightUp) {
+                camera._orientation = reading.orientation;
+            }
+        }
+    }
 
     // Orientation sensors for primary (back camera) & secondary (front camera)
     readonly property var _rotationValues: {
-        "primary": [270, 270, 90, 180 ,0],
-        "secondary": [90, 90, 270, 180, 0]
-    }
-
-    NumberAnimation on controlsRotation { running: camera._orientation === 1; to: -90;   duration: 200 }
-    NumberAnimation on controlsRotation { running: camera._orientation === 2; to: 90; duration: 200 }
-    NumberAnimation on controlsRotation { running: camera._orientation === 3; to: 180; duration: 200 }
-    NumberAnimation on controlsRotation { running: camera._orientation === 4; to: 0;  duration: 200 }
-
-    focus: true
-
-    Rectangle {
-        parent: window
-        anchors.fill: parent
-        z: -1
-        color: "black"
+        "primary": [270, 270, 90, 180 ,0, 270, 270], //Uses orientation sensor value 0-6
+        "secondary": [90, 90, 270, 180, 0, 90, 90],
+        "ui": [0, 90, 0, 0, 270, 0, 0, 0, 180] //Uses enum value 1,2,4,8
     }
 
-    FSOperations {
-        id: fsOperations
-    }
-
-    VideoOutput {
-        id: captureView
-        anchors.fill: parent
-        source: camera
-        rotation: camera.position == Camera.FrontFace ? 0 : 180
-        orientation: camera.orientation
-        onOrientationChanged: {
-            console.log(orientation);
+    readonly property int viewfinderOrientation: {
+        var rotation = 0
+        switch (orientation) {
+        case Orientation.Landscape: rotation = 90; break;
+        case Orientation.PortraitInverted: rotation = 180; break;
+        case Orientation.LandscapeInverted: rotation = 270; break;
         }
-    }
-
-    GridOverlay {
-        aspect: settings.global.captureMode == "image" ? ratio(camera.imageCapture.resolution) : ratio(camera.videoRecorder.resolution)
 
-        function ratio(resolution) {
-            return resolution.width / resolution.height
-        }
+        return (720 + camera.orientation + rotation) % 360
     }
 
-    Slider {
-        id: zoomSlider
-        anchors.horizontalCenter: parent.horizontalCenter
-        anchors.bottom: parent.bottom
-        width: parent.width * 0.75
-        minimumValue: 1
-        maximumValue: camera.maximumDigitalZoom
-        value: camera.digitalZoom
-        stepSize: zoomStepSize
-        rotation: {
-            // Zoom slider should be slide up to zoom in
-            if (camera._orientation  === 1 ) return -180
-            else if (camera._orientation === 2) return 0
-            else return controlsRotation }
-
-        onValueChanged: {
-            if (value != camera.digitalZoom) camera.digitalZoom = value;
-        }
+    RotationAnimation on controlsRotation { running: camera._orientation === OrientationReading.TopUp; to: 270;   duration: 200 }
```

Comment:
Add "direction: RotationAnimation.Shortest" to all of these to make transitions smooth in all cases.